src/sspi/negotiate.rs

```diff
@@ -43,29 +41,35 @@ pub trait ProtocolConfig: Debug {
 pub struct NegotiateConfig {
     pub protocol_config: Box<dyn ProtocolConfig + Send>,
     pub package_list: Option<String>,
+    pub hostname: String,
+    pub network_client_generator: Box<dyn NetworkClientFactory>,
 }
 
 impl NegotiateConfig {
-    pub fn new(protocol_config: Box<dyn ProtocolConfig + Send>, package_list: Option<String>) -> Self {
+    pub fn new(
+        protocol_config: Box<dyn ProtocolConfig + Send>,
+        package_list: Option<String>,
+        hostname: String,
+        get_network_client: Box<dyn NetworkClientFactory>,
+    ) -> Self {
         Self {
             protocol_config,
             package_list,
+            hostname,
+            network_client_generator: get_network_client,
```

Comment:
I feel it would improve clarity to keep the same terminology at both places.
```suggestion
            network_client_factory: get_network_client,
```